hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -315,23 +352,49 @@ def post_apply(self, updated: t.Dict = {}):
         table = dynamodb.Table(self.data_store_table)
 
         for update in updated.values():
-            row = update.as_csv_row()
+            row: t.List[str] = update.as_csv_row()
             # example row format: ('<raw_indicator>', '<indicator-id>', '<descriptor-id>', '<time added>', '<space-separated-tags>')
             # e.g (10736405276340','096a6f9...064f', '1234567890', '2020-07-31T18:47:45+0000', 'true_positive hma_test')
-            if PDQSignalMetadata(
-                signal_id=int(row[1]),
+            new_tags = row[4].split(" ") if row[4] else []
+
+            metadata = PDQSignalMetadata.get_from_signal_and_ds_id(
+                table,
+                int(row[1]),
+                S3ThreatDataConfig.SOURCE_STR,
+                str(self.privacy_group),
+            )
+
+            if metadata:
+                new_pending_opinion_change = self.get_new_pending_opinion_change(
+                    metadata, new_tags
+                )
+            else:
+                # If this is a new indicator without metadata we should have no pending opinion about it
+                new_pending_opinion_change = PendingOpinionChange.NONE
+
+            metadata = PDQSignalMetadata(
+                signal_id=row[1],
                 ds_id=str(self.privacy_group),
                 updated_at=datetime.now(),
                 signal_source=S3ThreatDataConfig.SOURCE_STR,
                 signal_hash=row[0],  # note: not used by update_tags_in_table_if_exists
-                tags=row[4].split(" ") if row[4] else [],
-            ).update_tags_in_table_if_exists(table):
+                tags=new_tags,
+                pending_opinion_change=new_pending_opinion_change,
+            )
+            if metadata.update_tags_in_table_if_exists(table):
```

Comment:
nit: having two update calls is less than ideal but not going to block on it (can you add a #TODO comment)